AutomatedTesting/Gem/CMakeLists.txt

```diff
@@ -6,9 +6,30 @@
 #
 #
 
-set(gem_path ${CMAKE_CURRENT_LIST_DIR})
+# Query the gem name from the gem.json file if possible
+# otherwise fallback to using AutomatedTesting
+o3de_find_ancestor_gem_root(gem_path gem_name "${CMAKE_CURRENT_SOURCE_DIR}")
+if (NOT gem_name)
+    set(gem_name "AutomatedTesting")
+endif()
+
+# Fallback to using the current source CMakeLists.txt directory as the gem root path
+if (NOT gem_path)
+    set(gem_path ${CMAKE_CURRENT_SOURCE_DIR})
+endif()
+
 set(gem_json ${gem_path}/gem.json)
+
+# Read the version field from the gem.json
+set(gem_version, "0.0.0")
+o3de_read_json_key(gem_version ${gem_json} "version")
+
 o3de_restricted_path(${gem_json} gem_restricted_path gem_parent_relative_path)
 
+o3de_pal_dir(pal_dir ${CMAKE_CURRENT_SOURCE_DIR}/Platform/${PAL_PLATFORM_NAME} "${gem_restricted_path}" "${gem_path}" "${gem_parent_relative_path}")
+
+ly_add_external_target_path(${CMAKE_CURRENT_SOURCE_DIR}/3rdParty)
+
```

Comment:
What do you think about putting this common code into a macro in `Gems.cmake` and calling it with something like:
```
o3de_gem_common_setup("<fall back gem name>", "<fall back version")
```

Reply:
That sounds good.